Include/fileutils.h

```diff
@@ -81,6 +81,18 @@ PyAPI_FUNC(int) _Py_EncodeLocaleEx(
 #ifndef Py_LIMITED_API
 PyAPI_FUNC(PyObject *) _Py_device_encoding(int);
 
+#if defined(MS_WINDOWS) || defined(__APPLE__)
+/* On Windows, the count parameter of read() is an int.
+ * macOS fails when reading or writing more than 2GB, see bpo-24658 */
+#   define _PY_READ_MAX  INT_MAX
+#   define _PY_WRITE_MAX INT_MAX
+#else
+#   define _PY_READ_MAX  PY_SSIZE_T_MAX
+#   define _PY_WRITE_MAX PY_SSIZE_T_MAX
+/* write() should truncate count to PY_SSIZE_T_MAX, but it's safer
```

Comment:
nitpick: you may align /* on define :-) and move the comment before #   define _PY_WRITE_MAX, not after.